sdk/test/NetStandard/UnitTests/Core/Credentials/DefaultInstanceProfileAWSCredentialsTests.cs

```diff
@@ -0,0 +1,373 @@
+﻿using Amazon.Runtime;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace UnitTests.NetStandard.Core.Credentials
+{
+    public sealed class DefaultInstanceProfileAWSCredentialsTests
+    {
+        // Helper: release the private semaphore so the instance can proceed.
+        private static void EnsureSemaphoreReleased(DefaultInstanceProfileAWSCredentials instance)
+        {
+            var field = typeof(DefaultInstanceProfileAWSCredentials)
+                .GetField("_credentialsSemaphore", BindingFlags.Instance | BindingFlags.NonPublic);
+            var sem = (SemaphoreSlim)field.GetValue(instance);
+            if (sem.CurrentCount == 0)
+            {
+                // Safe: call Release once to allow entry.
+                sem.Release();
+            }
+        }
+
+        // Helper: reset the private static failure flag between tests.
+        private static void ResetPreviousRefreshFailedFlag()
+        {
+            var field = typeof(DefaultInstanceProfileAWSCredentials)
+                .GetField("_previousRefreshFailed", BindingFlags.Static | BindingFlags.NonPublic);
+            field.SetValue(null, false);
+        }
+
+        [Fact]
+        public async Task GetCredentialsAsync_FetchesFromIMDS_WhenNoCached()
+        {
+            ResetPreviousRefreshFailedFlag();
+
+            var expectedAccessKey = "AK123";
+            var expectedSecret = "SK123";
+            var expectedToken = "T123";
+
+            var stub = new SimpleStubImds
+            {
+                CredentialsToReturn = new RefreshingAWSCredentials.CredentialsRefreshState(
+                    new ImmutableCredentials(expectedAccessKey, expectedSecret, expectedToken),
+                    DateTime.UtcNow.AddMinutes(30))
+            };
+
+            // lower refresh rate to speed test
+            var provider = DefaultInstanceProfileAWSCredentials.CreateTestDefaultInstanceProfileAWSCredentials(stub, TimeSpan.FromSeconds(5));
+
+            // fix semaphore (production code's semaphore is created with initialCount:0 in this branch)
```

Comment:
I don't understand the comment. The code does `initialCount: 1`. Also what is "this branch"? Note there are a few places where this comment exists.